vars/sonarExecuteScan.groovy

```diff
@@ -1,231 +1,88 @@
-                    credentialsId: config.sonarTokenCredentialsId,
-                    variable: 'SONAR_TOKEN'
-                )]){
-                    config.options.add("sonar.login=$SONAR_TOKEN")
-                    workerForSonarAuth(config)
+            // load certificates into cacerts file
+            loadCertificates([customTlsCertificateLinks: stepConfig.customTlsCertificateLinks, verbose: stepConfig.verbose])
+            // execute step
+            dockerExecute(
+                script: script,
+                dockerImage: config.dockerImage,
+                dockerWorkspace: config.dockerWorkspace,
+                dockerOptions: config.dockerOptions,
+            ) {
+                if(!fileExists('.git')) {
+                    utils.unstash('git')
                 }
-            }
-        }
-
-        if(isPullRequest()){
-            def workerForGithubAuth = worker
-            worker = { config ->
-                if(config.legacyPRHandling) {
-                    withCredentials([string(
-                        credentialsId: config.githubTokenCredentialsId,
-                        variable: 'GITHUB_TOKEN'
-                    )]){
-                        // support for https://docs.sonarqube.org/display/PLUG/GitHub+Plugin
-                        config.options.add('sonar.analysis.mode=preview')
-                        config.options.add("sonar.github.oauth=$GITHUB_TOKEN")
-                        config.options.add("sonar.github.pullRequest=${env.CHANGE_ID}")
-                        config.options.add("sonar.github.repository=${config.githubOrg}/${config.githubRepo}")
-                        if(config.githubApiUrl) config.options.add("sonar.github.endpoint=${config.githubApiUrl}")
-                        if(config.disableInlineComments) config.options.add("sonar.github.disableInlineComments=${config.disableInlineComments}")
-                        workerForGithubAuth(config)
-                    }
-                } else {
-                    // see https://sonarcloud.io/documentation/analysis/pull-request/
-                    config.options.add("sonar.pullrequest.key=${env.CHANGE_ID}")
-                    config.options.add("sonar.pullrequest.base=${env.CHANGE_TARGET}")
-                    config.options.add("sonar.pullrequest.branch=${env.CHANGE_BRANCH}")
-                    config.options.add("sonar.pullrequest.provider=${config.pullRequestProvider}")
-                    switch(config.pullRequestProvider){
-                        case 'GitHub':
-                            config.options.add("sonar.pullrequest.github.repository=${config.githubOrg}/${config.githubRepo}")
-                            break
-                        default: error "Pull-Request provider '${config.pullRequestProvider}' is not supported!"
+                withSonarQubeEnv(stepConfig.instance) {
+                    withCredentials(credentials) {
+                        sh "./piper ${STEP_NAME}"
```

Comment:
Here you don't use custom defaults or config file location either. Maybe you could use --stepConfig and pass the already resolved stepConfig map as JSON. But I am not sure if that will then ignore the credentials.